library/libwaku.nim

```diff
@@ -25,52 +25,34 @@ import
   ./waku_thread/inter_thread_communication/requests/ping_request,
   ./waku_thread/inter_thread_communication/waku_thread_request,
   ./alloc,
-  ./callback
+  ./ffi_types
 
 ################################################################################
 ### Wrapper around the waku node
 ################################################################################
 
-################################################################################
-### Exported types
-
-const RET_OK: cint = 0
-const RET_ERR: cint = 1
-const RET_MISSING_CALLBACK: cint = 2
-
-### End of exported types
-################################################################################
-
 ################################################################################
 ### Not-exported components
 
-template foreignThreadGc(body: untyped) =
-  when declared(setupForeignThreadGc):
-    setupForeignThreadGc()
-
-  body
+template checkLibwakuParams*(
+    ctx: ptr WakuContext, callback: WakuCallBack, userData: pointer
+) =
+  ctx[].userData = userData
 
-  when declared(tearDownForeignThreadGc):
-    tearDownForeignThreadGc()
+  if isNil(callback):
+    return RET_MISSING_CALLBACK
 
-proc handleRes[T: string | void](
-    res: Result[T, string], callback: WakuCallBack, userData: pointer
+proc handleSentToChannelRes(
```

Comment:
It seems as if this proc is no longer needed because we are handling the responses within the `waku_thread_request` module. I see some room for simplification by completely removing that one :)

Reply:
I did a small refactor. Should simplify a bit the code